include/aikido/robot/ros/RosRobot.hpp

```diff
@@ -18,21 +21,198 @@ AIKIDO_DECLARE_POINTERS(RosRobot)
   /// \param[in] urdf The path to the robot's URDF.
   /// \param[in] srdf The path to the robot's SRDF.
   /// \param[in] name The name of the robot.
   /// \param[in] retriever Retriever to access resources (urdf/srdf).
+  /// \param[in] rosControllerManagerServerName The name of the controller
+  ///            manager server.
+  /// \param[in] rosJointModeServerName The name of the joint
+  ///            mode controller actionlib client.
   RosRobot(
       const dart::common::Uri& urdf,
       const dart::common::Uri& srdf,
       const std::string name,
       const bool addDefaultExecutors,
       const dart::common::ResourceRetrieverPtr& retriever
-      = std::make_shared<aikido::io::CatkinResourceRetriever>());
+      = std::make_shared<aikido::io::CatkinResourceRetriever>(),
+      std::shared_ptr<::ros::NodeHandle> node = nullptr);
+
+  ///
+  /// Construct a new RosRobot as subrobot.
+  ///
+  /// \param[in] refSkeleton The metaskeleton defining the robot.
+  /// \param[in] rootRobot Pointer to parent robot
+  /// \param[in] collisionDetector Parent robot's collision detector
+  /// \param[in] collisionFilter Parent robot's collision filter
+  /// \param[in] name Unique Name of sub-robot
+  RosRobot(
+      dart::dynamics::ReferentialSkeletonPtr refSkeleton,
+      Robot* rootRobot,
+      dart::collision::CollisionDetectorPtr collisionDetector,
+      std::shared_ptr<dart::collision::BodyNodeCollisionFilter> collisionFilter,
+      const std::string name)
+    : Robot(refSkeleton, rootRobot, collisionDetector, collisionFilter, name)
+  {
+  }
 
   /// Destructor.
   virtual ~RosRobot() = default;
+
+  ///
+  /// Steps the ros robot (and underlying executors and subrobots) through time.
+  /// Call regularly to update the state of the ros robot.
+  ///
+  /// \param[in] timepoint The point in time to step to.
+  void step(const std::chrono::system_clock::time_point& timepoint) override;
+
+  ///
+  /// Registers a subset of the joints of the skeleton as a new RosRobot.
+  /// Must be disjoint from other subrobots.
+  ///
+  /// \param[in] metaSkeleton The referential skeleton corresponding to the
+  /// subrobot. \param[in] name Name of the subrobot.
+  RobotPtr registerSubRobot(
+      dart::dynamics::ReferentialSkeletonPtr refSkeleton,
+      const std::string& name) override;
+
+  ///
+  /// Stops the controller corresponding to this executor and deactivates
+  /// executor.
+  void deactivateExecutor() override;
+
+  ///
+  /// Loads the controller. Throws a runtime_error if controller cannot be
+  /// loaded. Add an (executor, controller, control mode) to the inactive
+  /// (executor, controller, control mode) list Releases DoFs held by executor
+  /// if held.
+  ///
+  /// \param[in] executor The Executor to add to the inactive list.
+  /// \param[in] controllerName The corresponding controller to load and add to
+  /// the inactive list. \param[in] controllerMode The corresponding control
```

Comment:
\param[in] and \return need to be on new lines. Note `make format` will often screw this up the first time.